### PR TITLE
chore(fastapi/starlette): adding patch testing

### PR DESCRIPTION
**Pull request by ZStriker19**
In preparation for https://github.com/DataDog/dd-trace-py/pull/3612 since in that PR Starlette patching is added to Fastapi patching.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: tests/contrib/fastapi/test_fastapi_patch.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    __integration_name__ = "fastapi"
+    __module_name__ = "fastapi"
+    __patch_func__ = patch
+    __unpatch_func__ = None
+
+    def assert_module_patched(self, fastapi):
+        self.assert_wrapped(fastapi.__init__)
+
+    def assert_not_module_patched(self, fastapi):
+        self.assert_wrapped(fastapi.__init__)
```
Comment (brettlangdon) [inline tests/contrib/fastapi/test_fastapi_patch.py:15]:
```suggestion
        self.assert_not_wrapped(fastapi.__init__)
```

File: tests/contrib/fastapi/test_fastapi_patch.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    def assert_not_module_double_patched(self, fastapi):
+        self.assert_wrapped(fastapi.__init__)
```
Comment (brettlangdon) [inline tests/contrib/fastapi/test_fastapi_patch.py:18]:
```suggestion
        self.assert_not_double_wrapped(fastapi.__init__)
```

File: tests/contrib/fastapi/test_fastapi_patch.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from ddtrace.contrib.fastapi import patch
```
Comment (brettlangdon) [inline tests/contrib/fastapi/test_fastapi_patch.py:1]:
```suggestion
from ddtrace.contrib.fastapi import patch, unpatch
```

File: tests/contrib/fastapi/test_fastapi_patch.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+from tests.contrib.patch import PatchTestCase
+
+
+class TestFastapiPatch(PatchTestCase.Base):
+    __integration_name__ = "fastapi"
+    __module_name__ = "fastapi"
+    __patch_func__ = patch
+    __unpatch_func__ = None
```
Comment (brettlangdon) [inline tests/contrib/fastapi/test_fastapi_patch.py:9]:
```suggestion
    __unpatch_func__ = unpatch
```

File: tests/contrib/starlette/test_starlette_patch.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+from ddtrace.contrib.starlette import patch
+from tests.contrib.patch import PatchTestCase
+
+
+class TestStarlettePatch(PatchTestCase.Base):
```
Comment (brettlangdon) [inline tests/contrib/starlette/test_starlette_patch.py:5]: same comments apply here

**Review by P403n1x87 — DISMISSED** [review]

**Review by brettlangdon — DISMISSED** [review]

File: tests/contrib/fastapi/test_fastapi_patch.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+from ddtrace.contrib.fastapi import patch, unpatch
```
Comment (brettlangdon) [inline tests/contrib/fastapi/test_fastapi_patch.py:1]:
```suggestion
from ddtrace.contrib.fastapi import patch
from ddtrace.contrib.fastapi import unpatch
```

**Review by mabdinur — APPROVED** [review]

**Review by Kyle-Verhoog — COMMENTED** [review]
@ZStriker19 looks like the fastapi tests are failing

**Review by P403n1x87 — APPROVED** [review]

File: tests/contrib/patch.py [hunk 7]
```diff
@@ -312,6 +312,12 @@ def test_import_patch(self):
             self.__patch_func__()
             self.assert_module_patched(module)
 
+            # import fastapi
+            # from ddtrace import patch
+
+            # patch(fastapi=True)
+            # self.assert_module_patched(fastapi)
+
```
Comment (brettlangdon) [inline tests/contrib/patch.py:320]: No longer needed, can we delete?

**Review by brettlangdon — APPROVED** [review]

File: tests/contrib/fastapi/test_fastapi_patch.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class TestFastapiPatch(PatchTestCase.Base):
+    __integration_name__ = "fastapi"
+    __module_name__ = "fastapi"
+    __patch_func__ = patch
+    __unpatch_func__ = unpatch
+
+    def assert_module_patched(self, fastapi):
+        self.assert_wrapped(fastapi.applications.FastAPI.__init__)
```
Comment (brettlangdon) [inline tests/contrib/fastapi/test_fastapi_patch.py:13]: We are missing testing for `serialize_response`.

https://github.com/DataDog/dd-trace-py/blob/1595f9f9d/ddtrace/contrib/fastapi/patch.py#L72-L72

**Review by brettlangdon — CHANGES_REQUESTED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**